### ISSUE TITLE
[question] `--svg`, convert picture to svg?

### ISSUE THREAD

**Issue by nschloe** [brucemiller/LaTeXML#2144]
I would like to use `latexmlpost` to convert whatever `<picture>`s are in `latexml`'s output to SVG files. I tried

```
latexmlpost out.xml --svg
```

which gives "No obvious problems", but doesn't produce SVG files either.

Any hint?

**Comment by dginev** [comment]
Thanks to @rzach for referencing this issue, let me circle back with a comment.

I think we have no configurable handle on how the SVGs are serialized @nschloe . Currently they're always deposited inline the XML/HTML output of latexml/latexmlpost. I can see the benefit of writing separate files, as certain services can benefit from as small as possible main HTML file (e.g. social media cards often have an indexing maxsize of 2MB, and will drop the page if it exceeds that limit).

I believe there was some simplicity in defaulting to inline serialization, as SVGs can contain various interactive elements, notably links via `\ref` or `\cite`, which would need to be tracked in cases one splits the final document via `--split`. So, if we were to enable a variant that serializes each SVG image into a standalone file, I think we would need to do that as late as possible in post-processing, likely as an extension to the `Split` or even `Writer` processor.

I'm marking this as an enhancement request, and pushing it a little down the milestone list.

**Comment by brucemiller** [comment]
Right, the `--svg` option simply specifies whether (or not) to try to convert pictures to SVG, but only to embed the SVG, not to separate it out as a separate file. And since, in general, the SVG can contain foreignObjects (html, math, links) as well as CSS,  classes, etc, its is most natural to embed.

I suppose a post-processor could separate them out, but would have to validate, possibly transform, to get a valid stand-alone SVG.  Likewise, you could write a command-line tool somewhat like `latexmlmath` that would take a document containing only a picture, transform it to html w/embedded svg, and extract out the svg portion.

Without some pretty compelling use cases, though, this isn't a terribly high priority.
